### PR TITLE
feat(Kick): add presence

### PR DESCRIPTION
**Pull request by Extiriority**
## Description 
A rich presence addon for Kick, the most rewarding gaming and livestreaming platform. resolves https://github.com/PreMiD/Presences/issues/7200

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</>

</details>

### CODE REVIEW

File: websites/K/Kick/presence.ts [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+							buttons: [ButtonData, ButtonData?];
+						const streamer =
+							document.querySelector(".stream-username").textContent;
+						if (document.querySelector(".odometer-value")) {
+							state = `Watching: ${streamer}`;
+							smallImageKey = Assets.Live;
+							buttons = [
+								{
+									label: "Watch stream",
+									url: `https://kick.com/${streamer}`,
```
Comment (DarkVillager) [inline websites/K/Kick/presence.ts:146]: Couldn't `href` do the job here?

File: websites/K/Kick/presence.ts [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+								},
+							];
+						} else {
+							state = `Viewing: ${streamer}`;
+							smallImageKey = Assets.Viewing;
+							buttons = [
+								{
+									label: "View streamer",
+									url: `https://kick.com/${streamer}`,
```
Comment (DarkVillager) [inline websites/K/Kick/presence.ts:155]: Ditto

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/K/Kick/presence.ts (hunk 1 above)
Comment (Extiriority) [reply websites/K/Kick/presence.ts:146]: do you mean like  document.location.href?

Comment (DarkVillager) [reply websites/K/Kick/presence.ts:146]: > do you mean like document.location.href?

yes

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/K/Kick/presence.ts [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+	if (buttons) presenceData.buttons = buttons;
+	if (smallImageKey) presenceData.smallImageKey = smallImageKey;
+	if (state) presenceData.state = state;
+
+	if (!(await presence.getSetting<boolean>("details"))) {
+		presenceData.details = "Browsing Kick...";
+		presenceData.state = null;
+		presenceData.largeImageKey = Assets.Logo;
+		presenceData.smallImageKey = null;
+		presenceData.buttons = null;
```
Comment (theusaf) [inline websites/K/Kick/presence.ts:32]: Prefer `delete` over setting properties to null

File: websites/K/Kick/presence.ts [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+									label: "Watch stream",
+									url: link,
+								},
+							];
+						} else {
+							state = `Viewing: ${streamer}`;
+							smallImageKey = Assets.Viewing;
+							buttons = [
+								{
+									label: "View streamer",
```
Comment (theusaf) [inline websites/K/Kick/presence.ts:156]: Ditto

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by DarkVillager — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by theusaf — APPROVED** [review]

**Pull request merged**